app/javascript/components/patient/history/History.js

```diff
@@ -0,0 +1,185 @@
+import React from 'react';
+import { PropTypes } from 'prop-types';
+import { Button, Card, Col, Row } from 'react-bootstrap';
+import ReactTooltip from 'react-tooltip';
+import axios from 'axios';
+
+import DeleteDialog from '../../util/DeleteDialog';
+import reportError from '../../util/ReportError';
+import { formatTimestamp, formatRelativePast } from '../../../utils/DateTime';
+
+class History extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      loading: false,
+      editMode: false,
+      showDeleteModal: false,
+      comment: props.history.comment,
+    };
+  }
+
+  handleChange = event => {
+    this.setState({ [event.target.id]: event.target.value });
+  };
+
+  toggleEditMode = () => {
+    this.setState({ editMode: true });
+  };
+
+  toggleEditMode = () => {
+    let current = this.state.editMode;
+    this.setState({ editMode: !current, comment: this.props.history.comment });
+  };
+
+  handleEditSubmit = () => {
+    this.setState({ loading: true }, () => {
+      axios.defaults.headers.common['X-CSRF-Token'] = this.props.authenticity_token;
+      axios
+        .post(window.BASE_PATH + '/histories/' + this.props.history.id + '/edit', {
+          patient_id: this.props.history.patient_id,
+          comment: this.state.comment,
+        })
+        .then(() => {
+          location.reload(true);
+        })
+        .catch(error => {
+          reportError(error);
+        });
+    });
+  };
+
+  toggleDeleteModal = () => {
+    let current = this.state.showDeleteModal;
+    this.setState({
+      showDeleteModal: !current,
+      delete_reason: null,
+      delete_reason_text: null,
+    });
+  };
+
+  handleDeleteSubmit = () => {
+    let deleteReason = this.state.delete_reason;
+    if (deleteReason === 'Other' && this.state.delete_reason_text) {
+      deleteReason += ': ' + this.state.delete_reason_text;
+    }
+
+    this.setState({ loading: true }, () => {
+      axios.defaults.headers.common['X-CSRF-Token'] = this.props.authenticity_token;
+      axios
+        .post(window.BASE_PATH + '/histories/' + this.props.history.id + '/delete', {
+          patient_id: this.props.history.patient_id,
+          delete_reason: deleteReason,
+        })
+        .then(() => {
+          location.reload(true);
+        })
+        .catch(error => {
+          reportError(error);
+        });
+    });
+  };
+
+  renderEditMode() {
+    return (
+      <React.Fragment>
+        <textarea
```

Comment:
Do we want to add a limit to this? I believe SQL only allows strings of 8,000 characters. The call fails if you try to send more than that to the backend.

Reply:
that would be a question for @holmesie.  currently we don't cap in the add comment text input so if we do one we should do the other

Reply:
We did at some point, but it looks like that was lost when this was converted to a React component.  This field is actually not a `string` in the db, it's a `text` - so it does support more than 8k. We limit it to 10k characters in the model validation. This is great feedback, and it's never a bad idea to add that extra bit of validation to the front end.

Reply:
So what do you want to limit this at?

Reply:
Well, we limit it to 10k characters on the backend, so 10k is probably a good choice here as well.

Reply:
Added here: https://github.com/SaraAlert/SaraAlert/pull/793/commits/ddeeb2c07f876d766273b0a1823b00177975d4e4